Revised Technical Requirements for accessibility, clarity and to remo…

ceford opened the pull request:
### **User description**
…ve obsolete content. Content removed includes a link to a tutorial for IIS6 and IIS7.


___

### **PR Type**
Documentation


___

### **Description**
- Updated technical requirements for Joomla! 5.x and 4.x.

- Removed obsolete content and outdated references.

- Improved clarity and accessibility of documentation.

- Added additional information on software requirements.


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>technical-requirements.md</strong><dd><code>Revise Joomla! technical requirements documentation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/get-started/technical-requirements.md

<li>Updated technical requirements tables for Joomla! 5.x and 4.x.<br> <li> Removed obsolete content, including outdated IIS links.<br> <li> Added detailed additional information on software requirements.<br> <li> Improved formatting and accessibility of the document.


</details>


  </td>
  <td><a href="https://github.com/joomla/Manual/pull/393/files#diff-552d80425038f200ae721b924cd92b5f043c548342805a47aebb235c69344e46">+33/-39</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

HLeithner commented:
I would like to keep the footnotes please

ceford commented:
I made some changes but got into a muddle with pushing. However...

I inserted Footnotes for Recommended and Minimum. The others don't make any sense. Why are we putting Windows installation instructions in this document? I left in links to the main sites for the recommended software but why we include those here does not seem so logical to me. 

In general, I think we should avoid footnotes wherever possible. They push some essential information to the bottom of the page, which might be a long way down and is quite different from what happens in printed books. In this case think it would be preferable not to use Markdown footnotes but just to type in the footnotes beneath the table.

Take a look and come back if you would like more changes.

ceford commented:
From an AI Source:

Footnotes are generally considered bad practice on web pages because they disrupt the reading flow, can be difficult to access on mobile devices, and can be problematic for screen readers, making them less user-friendly compared to other methods like in-text citations or pop-up tooltips; most experts recommend avoiding footnotes on websites unless absolutely necessary and opting for alternative methods to provide additional information. 

Key reasons why footnotes are not ideal for websites:

* Navigation disruption: Unlike printed pages where footnotes are easily accessible at the bottom, on a website, users might have to scroll significantly to reach the footnote, interrupting their reading flow. 
* Accessibility issues: Screen readers may not interpret footnotes correctly, leaving users unaware of the additional information. 
* Mobile challenges: Clicking on small footnote links on a mobile device can be cumbersome. 

Alternatives to footnotes on websites:

* In-text citations: Embed citations directly within the text, often with a link to the source. 
* Tooltips: Hovering over a highlighted word or phrase can reveal additional information in a pop-up window. 
* Side notes: Display additional information alongside the main text, often in a smaller font.

HLeithner commented:
I rethought the thing again and your changes.
1. I would say the footnotes you re-added makes less sense and was better in your first change
2. What I'm missing now are the notes for the modules which are needed, you collected them in the footer but actually they can be different between versions.

ceford commented:
I moved the footnotes to the sentence before the tables.

The module notes are inconsistent. The only difference is whether or not zlib is in the list. It was in the 4.x list for the 4.4 and 5.0 releases and then disappeared. So is it or is it not required for 4.x? My 4.4 test installation shows Zlib Enabled / Yes in the PHP Settings. What about other modules such as Mbstring? It seems to me that we can mention any specific module requirement for any new Joomla version in the Additional Information rather than an extra column in the table.

HLeithner commented:
My point on having it directly at the version is to have all requirements at one place for a version. It doesn't need to be a table, if you have a nicer way to display it it's also fine for me.

Only thing I have in mind is to have one version in one place, if you add it to the additional information you look at 5.x and have to scroll over the other version to find out what's needed for your version. I think that's not the default use case for this document. 

I think you are come here because you downloaded or want to install joomla 5.4.3 and need to know what's required to run it, in this case I'm not interested to scrape the full page to get this information. 

ymmv

ceford commented:
For this version I have moved the PHP Module Requirements to a line following each table (pity it is not possible to merge cells) to get a reasonable combination of layout and information.

In the PHP additional information section I added a link to the FPA. Is that a good/bad idea?

HLeithner commented:
Might be a bit better, but actually for Apache and MS IIS the information are still somewhere else.
Also it's a bit strange to add the "[Forum Post Assistant](https://forumpostassistant.github.io/docs/)" from my point of view this makes no sense as it's not required to run joomla.

I would move this optional requirement above:
> [Apache](https://httpd.apache.org/) requires the mod_rewrite extension to use SEO URLs.

I would reparse this by removing "may" because you need PHP ;-) but you don't need mysql and the Urlrewriter is optional as the mod_rewrite in apache
> [Microsoft IIS](https://www.iis.net/) may need the following: PHP, MySQL, Microsoft [URL Rewrite Module](https://learn.iis.net/page.aspx/460/using-url-rewrite-module/) within IIS Manager for SEO URLs only, FastCGI

I would remove the sentence about the removed ext/pgsql
> [PostgreSQL](https://www.postgresql.org/) is open source and available in numerous editions. ext/pgsql support in PHP has been removed. It now uses the PostgreSQL PDO Driver.

For nginx I would add atleast a sentence what it is ;-)

HLeithner commented on docs/get-started/technical-requirements.md:
```diff
@@ -4,53 +4,51 @@ sidebar_position: 1
+| Microsoft IIS      | 10              | 10          |
 
-### Requirements for older Versions
+**Required PHP Modules:** json, simplexml, dom, zlib, gd, mysqlnd or pdo_mysql or pdo_pgsql
```
```suggestion
**Required PHP Modules:** json, simplexml, dom, zlib, gd, mysqlnd or pdo_mysql or pdo_pgsql
**Optional Apache Modules:** _mod_rewrite_ extension to use SEO URLs
**Optional Microsoft IIS Modules:** [URL Rewrite Module](https://learn.iis.net/page.aspx/460/using-url-rewrite-module/) to use SEO URLs

```

HLeithner commented on docs/get-started/technical-requirements.md:
```diff
@@ -4,53 +4,51 @@ sidebar_position: 1
+| Microsoft IIS      | 10              | 8           |
 
-## Footnotes
+**Required PHP Modules:** json, simplexml, dom, zlib, gd, mysqlnd or pdo_mysql or pdo_pgsql
```
```suggestion
**Required PHP Modules:** json, simplexml, dom, zlib, gd, mysqlnd or pdo_mysql or pdo_pgsql
**Optional Apache Modules:** _mod_rewrite_ extension to use SEO URLs
**Optional Microsoft IIS Modules:** [URL Rewrite Module](https://learn.iis.net/page.aspx/460/using-url-rewrite-module/) to use SEO URLs
```

HLeithner commented on docs/get-started/technical-requirements.md:
```diff
@@ -4,53 +4,51 @@ sidebar_position: 1
+## Additional Information
 
-[^4]: In order to use SEO URLs, you will need to have the Apache mod_rewrite extension installed.
+- [PHP](https://www.php.net) The [Forum Post Assistant](https://forumpostassistant.github.io/docs/) can help to diagnose PHP problems with an installed Joomla instance.
+- [MySQL](https://www.mysql.com) is available in several editions. It is normal to install a GPL Community Edition for your software platform.
+- [MariaDB](https://mariadb.com) is also available in several editions. It is normal to install a free Community Edition for your software platform.
+- [PostgreSQL](https://www.postgresql.org/) is open source and available in numerous editions. ext/pgsql support in PHP has been removed. It now uses the PostgreSQL PDO Driver.
+- [Apache](https://httpd.apache.org) requires the mod_rewrite extension to use SEO URLs.
+- [Nginx](https://www.nginx.com/resources/)
+- [Microsoft IIS](https://www.iis.net) may need the following:  PHP, MySQL, Microsoft [URL Rewrite Module](https://learn.iis.net/page.aspx/460/using-url-rewrite-module/) within IIS Manager for SEO URLs only, FastCGI
```
```suggestion
- [PHP](https://www.php.net) A popular general-purpose scripting language that is especially suited to web development.
- [MySQL](https://www.mysql.com) is available in several editions. It is normal to install a GPL Community Edition for your software platform.
- [MariaDB](https://mariadb.com) is also available in several editions. It is normal to install a free Community Edition for your software platform.
- [PostgreSQL](https://www.postgresql.org/) is open source and available in numerous editions.
- [Apache](https://httpd.apache.org) HTTP Server Project is an effort to develop and maintain an open-source HTTP server for modern operating systems including UNIX and Windows.
- [Nginx](https://nginx.org) is an HTTP web server, reverse proxy, content cache, load balancer.
- [Microsoft IIS](https://www.iis.net) (Internet Information Services) for Windows® Server is a flexible, secure and manageable Web server for hosting anything on the Web.
```

HLeithner reviewed (commented):
I added my suggestions

ceford commented:
Added your suggestions. The Preview looks good to me.

HLeithner commented on docs/get-started/technical-requirements.md:
```diff
@@ -4,53 +4,55 @@ sidebar_position: 1
+| Nginx              | 1.25            | 1.21        |
+| Microsoft IIS      | 10              | 10          |
+
+**Required PHP Modules:** json, simplexml, dom, zlib, gd, mysqlnd or pdo_mysql or pdo_pgsql<br>
+**Optional Apache Modules:** _mod_rewrite_ extension to use SEO URLs<br>
```
```suggestion
**Required PHP Modules:** json, simplexml, dom, zlib, gd, mysqlnd or pdo_mysql or pdo_pgsql  
**Optional Apache Modules:** _mod_rewrite_ extension to use SEO URLs  
```
Line breaks in markdown are done with spaces at the end `Text  `

HLeithner commented on docs/get-started/technical-requirements.md:
```diff
@@ -4,53 +4,55 @@ sidebar_position: 1
+| Nginx              | 1.18            | 1.10        |
+| Microsoft IIS      | 10              | 8           |
+
+**Required PHP Modules:** json, simplexml, dom, zlib, gd, mysqlnd or pdo_mysql or pdo_pgsql<br>
+**Optional Apache Modules:** _mod_rewrite_ extension to use SEO URLs<br>
```
```suggestion
**Required PHP Modules:** json, simplexml, dom, zlib, gd, mysqlnd or pdo_mysql or pdo_pgsql  
**Optional Apache Modules:** _mod_rewrite_ extension to use SEO URLs  
```

HLeithner commented:
Finally I think it's ok for now. thanks

ceford commented:
Thank you. For line breaks I was going on this: https://www.markdownguide.org/basic-syntax/#line-breaks
